torch/testing/_internal/common_fsdp.py

```diff
@@ -1374,15 +1374,27 @@ class FullyShardPatch(Enum):
     def decorator(func):
         @wraps(func)
         def wrapper(*args, **kwargs):
+            original_fully_shard = torch.distributed._composable.fsdp.fully_shard
             for fully_shard_patch in FullyShardPatch:
                 if fully_shard_patch != FullyShardPatch.EAGER and not has_triton():
                     warnings.warn("Inductor on GPU needs Triton and recent GPU arch")
                     continue
+                imported_fully_shard = (
+                    f"{func.__module__}.{original_fully_shard.__name__}"
+                )
                 with mock.patch(
-                    f"{func.__module__}.{torch.distributed._composable.fsdp.fully_shard.__name__}",
+                    imported_fully_shard,
                     fully_shard_patch.value,
                 ):
                     func(*args, **kwargs)
+                # mock.patch.__exit__ does not work with multi-thread
+                # thread 1 set func.__module__[fully_shard]
```

Comment:
Do the brackets in the `[fully_shard]` notation have special meaning? Or is it basically just saying setting `{func.__module__}.fully_shard`?